Add Image Link Updater plugin

andy51002000 opened the pull request:
This PR adds the Image Link Updater plugin.

- Repo: https://github.com/andy51002000/obsidian-image-link-updater
- Description: Automatically updates markdown image links when image files are renamed or moved in the vault.
- Author: Andy Lai
- License: AGPL-3.0-or-later
- Version: 1.3.0

joethei commented:
Remove the `main.js` file from the repo and add it to the `.gitignore`, release assets should not be mixed with source assets.

[type FileManagerLike = {](https://github.com/andy51002000/obsidian-image-link-updater/blob/65165dd021d9fab818addc89c47ed2bf0bfa8707/main.ts#L3-L7), [private getFileManager(): FileManagerLike | null {](https://github.com/andy51002000/obsidian-image-link-updater/blob/65165dd021d9fab818addc89c47ed2bf0bfa8707/main.ts#L326-L332)
FileManager is a public class, this code is unecessary.

andy51002000 commented:
@joethei Please help review the code.  I've removed the main.js file from the repo and add it to the .gitignore. I also removed the unecessary FileManager code.
